tfe/resource_tfe_workspace_test.go

```diff
@@ -366,6 +367,67 @@ func TestAccTFEWorkspace_updateTriggerPrefixes(t *testing.T) {
 	})
 }
 
+func TestAccTFEWorkspace_updateTriggerPatterns(t *testing.T) {
+	workspace := &tfe.Workspace{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			// Create trigger prefixes first so we can verify they are being removed if we introduce trigger patterns
+			{
+				Config: testAccTFEWorkspace_triggerPrefixes(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_prefixes.#", "2"),
+				),
+			},
+			{
+				Config: testAccTFEWorkspace_triggerPatterns(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEWorkspaceExists(
+						"tfe_workspace.foobar", workspace),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_patterns.#", "2"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_patterns.0", "/modules/**/*"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_patterns.1", "/**/networking/*"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_prefixes.#", "0"),
+				),
+			},
+
+			{
+				Config: testAccTFEWorkspace_updateEmptyTriggerPatterns(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEWorkspaceExists("tfe_workspace.foobar", workspace),
+					testAccCheckTFEWorkspaceAttributes(workspace),
+					resource.TestCheckResourceAttr("tfe_workspace.foobar", "trigger_patterns.#", "0"),
+				),
+			},
+		},
+	})
+}
```

Comment:
So in this test we:
1. create prefixes
2. create patterns and validate prefixes are no longer there
3. empty patterns.

I think we should do the opposite as well
1. create patterns
2. create prefixes and validate patterns are no longer there
3. empty prefixes

Reply:
We should also do an update call on patterns twice.

1. create a new resource with patterns
2. update the existing resource with new patterns